cfgov/unprocessed/js/modules/multi-select.js

```diff
@@ -0,0 +1,494 @@
+      } );
+
+      this.search = $.create( 'input', {
+        className:   'cf-multi-select_search',
+        type:        'text',
+        placeholder: self.placeholder || 'Choose up to five',
+        inside:      self.header
+      } );
+
+      this.fieldset = $.create( 'fieldset', {
+        className:  'cf-multi-select_fieldset',
+        visibility: 'hidden'
+      } );
+
+      this.list = $.create( 'ul', {
+        className: 'list__unstyled cf-multi-select_options',
+        inside:    self.fieldset
+      } );
+
+      this.optionData.forEach( function( option ) {
+        var li = $.create( 'li', {
+          'data-option': option.value
+        } );
+
+        $.create( 'input', {
+          id:        option.value,
+          value:     option.value,
+          type:      'checkbox',
+          name:      self.name,
+          class:     'cf-input cf-multi-select_checkbox',
+          inside:    li
+        } );
+
+        $.create( 'label', {
+          for:       option.value,
+          innerText: option.text,
+          className: 'cf-multi-select_label',
+          inside:    li
+        } );
+
+        self.list.appendChild( li );
+      } );
+
+      // Write our new markup to the DOM
+      this.container.appendChild( self.header );
+      this.container.appendChild( self.fieldset );
+    },
+
+    get selected() {
+      return this.index > -1;
+    },
+
+    open: function() {
+      this.container.classList.add( 'active' );
+      this.fieldset.setAttribute( 'visibility', 'visible' );
+    },
+
+    close: function() {
+      this.container.classList.remove( 'active' );
+      this.fieldset.setAttribute( 'visibility', 'hidden' );
+      this.index = -1;
+    },
+
+    next: function() {
+      this.highlight( 'next' );
+    },
+
+    previous: function() {
+      this.highlight( 'prev' );
+    },
+
+    // Should not be used, highlights specific item without any checks!
+    highlight: function( direction ) {
+      var self = this;
+      var count = this.filtered.length;
+      var value;
+      var item;
+
+      if ( direction === 'next' && this.index < count - 1 ) {
+        this.index += 1;
+      } else if ( direction === 'prev' && this.index > -1 ) {
+        this.index -= 1;
+      }
+
+      if ( this.index > -1 ) {
+        var value = this.filtered[ this.index ].value;
+        var item = this.list.querySelector( '[data-option="' + value + '"]' );
+        var input = $( item ).querySelector( 'input' );
+
+        self.skipBlur = true;
+        $( input ).focus();
+      } else {
+        self.skipBlur = false;
+        this.search.focus();
+      }
+    },
+
+    updateSelections: function( value ) {
+      var self = this;
+      var option = valInArray( self.optionData, value ).item ||
+                   self.optionData[self.index];
+
+      if ( option ) {
+        var inSelections = valInArray( self.selections, option.value );
+        var li;
+
+        if ( inSelections ) {
```

Comment:
I don't think that'll work right, because there's an else on L:191 for when the option isn't in the array of selections